### PR TITLE
Implements API Search Box in Dashboard

### PR DESCRIPTION
**Pull request by josemigallas**
**What this PR does / why we need it**:

Adds an input box for filtering services in the Dashboard
screen shot 2018-10-05 at 18 14 30

**Which issue(s) this PR fixes** 

https://issues.jboss.org/browse/THREESCALE-1315

**Special notes for your reviewer**:
Todos:
- [x] Verify performance (300 services, cpu slowdown x4 OK)
- [x] Tests
- [x] Styles good enough?

### CODE REVIEW

File: app/javascript/src/Dashboard/styles/dashboard.scss [hunk 1]
```diff
@@ -0,0 +1,43 @@
+$font-primary: #464646;
```
Comment (thomasmaas) [inline app/javascript/src/Dashboard/styles/dashboard.scss]: all colours are defined in https://github.com/3scale/porta/blob/dashboard-links/app/assets/stylesheets/provider/_colors.scss#L1. Try to pick one that makes sense instead of adding a new color.

File: app/javascript/src/Dashboard/styles/dashboard.scss [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+.ApiFilter {
+
+  float: left;
+  width: auto;
+  border: 1px solid #d1d1d1;
```
Comment (thomasmaas) [inline app/javascript/src/Dashboard/styles/dashboard.scss]: same for all hard coded width values: they are defined in https://github.com/3scale/porta/blob/dashboard-links/app/assets/stylesheets/provider/_typography.scss#L1

File: app/javascript/src/Dashboard/styles/dashboard.scss [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    width: auto;
+    padding: 0.5rem;
+  }
+
+  ::-webkit-input-placeholder {
+    font-style: normal;
+    font-weight: 600;
+    color: $font-primary;
+  }
+  ::-moz-placeholder {
```
Comment (thomasmaas) [inline app/javascript/src/Dashboard/styles/dashboard.scss]: placeholder styles are defined in https://github.com/3scale/porta/blob/dashboard-links/app/assets/stylesheets/provider/_forms.scss#L481

File: app/models/account.rb [hunk 4]
```diff
@@ -135,6 +135,9 @@ def schedule_backend_sync_worker
   has_many :sso_authorizations, through: :users
   has_many :user_sessions, through: :users
 
+  has_many :services
```
Comment (josemigallas) [inline app/models/account.rb]: How did this get here? @didierofrivia @thomasmaas

File: app/views/provider/admin/dashboards/show.html.slim [hunk 5]
```diff
@@ -23,9 +24,9 @@
     section.DashboardSection.DashboardSection--audience class=('DashboardSection--wide' unless can?(:manage, :plans))
       - if current_user.accessible_services.size > 1
         header.DashboardSection-header.DashboardSection-header--extended
-          h1.DashboardSection-title All API's
+          = render 'services_filter'
           = render 'services_navigation', api_docs: current_account.api_docs_services,
-                                          applications: current_account.provided_cinstances,
+                                          applications: current_account.cinstances,
```
Comment (hallelujah) [inline app/views/provider/admin/dashboards/show.html.slim]: Why this change?

Comment (josemigallas) [reply app/views/provider/admin/dashboards/show.html.slim]: This is wrong, rebase debris
